### DISCUSSION TITLE
Microsoft Identity Platform Authentication in Blazor Web Application

### DISCUSSION THREAD

**Discussion by giscus** [VladislavAntonyuk/vladislavantonyuk.github.io#69] [Announcements]
# Microsoft Identity Platform Authentication in Blazor Web Application

Adding Microsoft Entra ID B2C in .NET 8 Blazor WebApp.

https://vladislavantonyuk.github.io/articles/Microsoft-Identity-Platform-Authentication-in-Blazor-Web-Application/

**Comment by gabrewer** [comment]
Thank you for the great article.  Your template will be great for many trying to integrate the .Net Blazor web app with ADB2C.  I found your site and template while digging into an issue I am having with the call to MicrosoftIdentity/Account/SignOut  creating a loop.  I found this github issue https://github.com/dotnet/aspnetcore/issues/52245 which led me to your response in https://github.com/dotnet/aspnetcore/issues/51202.  I was hoping that your template would lead me to a solution, but I am having the same problem.  I am wondering if there was an issue with the Release bits vs RC2 or if there is something wrong with my app registration.  Is the SignOut like working for you?

**Reply by VladislavAntonyuk** [reply]
Add `builder.Services.AddRazorPages();` in Program.cs. `Microsoft.Identity.Web.UI` still uses razor pages.

I will update templates a bit later

**Reply by gabrewer** [reply]
Thank you for the quick response, it worked great.

**Comment by ham187** [comment]
This is great than you, is there anyway to get graph API working?

**Reply by VladislavAntonyuk** [reply]
yes, see the example here: https://github.com/VladislavAntonyuk/WorldExplorer/blob/main/src/WebApp/Infrastructure/ServiceExtensions.cs#L45

**Reply by ham187** [reply]
Thank you, I tried to get it working with your b2c template but no success.. :(

Any chance you could please, please, make a template with b2c and graph api, a lot of people are looking for it :)

Thank you!!!

**Reply by Deineris** [reply]
Is it possible to make it work for the interactive rendering? This whole b2c registration is a little bit confusing. The template have client and server folders, so I assume it is more like WASM security schema? Or is there something else in hybrid rendering mode? I assume server-side should work similar to the .net 7, but is there a template somwhere for blazor web app in conjunction with azure b2c?

**Reply by VladislavAntonyuk** [reply]
I created template for Blazor WebApp with Server Interaction Mode only. For now there is no template for both Server and Client Interactions (Auto mode).
Microsoft scheduled it for the .NET 9.

**Reply by Deineris** [reply]
So my only option if I want to have b2c as an auth mechanism is to use olders versions or convert to blazor server, right?

**Reply by VladislavAntonyuk** [reply]
You can use the latest .NET 8 but only with Server interaction mode. Feel free to use my template and see how it works
